sw/host/tests/chip/jtag/src/main.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::time::Duration;
+use structopt::StructOpt;
+use regex::Regex;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::io::jtag::{JtagParams, JtagTap};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::{execute_test};
+use opentitanlib::transport::common::fpga::ClearBitstream;
+use opentitanlib::uart::console::{ExitStatus, UartConsole};
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams
+}
+
+// RAII-style struct that clears the FPGA's bitstream when it goes out of scope.
+struct ScopedBitstream<'a> {
+    transport: &'a TransportWrapper,
+}
+
+impl Drop for ScopedBitstream<'_> {
+    fn drop(&mut self) {
+        self.transport
+            .dispatch(&ClearBitstream {})
+            .expect("Should dispatch ClearBitstream command successfully");
+    }
+}
+
+fn reset(transport: &TransportWrapper, strappings: &[&str], reset_delay: Duration) -> Result<()> {
+    log::info!("Resetting target...");
+    for strapping in strappings.iter() {
+        transport.apply_pin_strapping(strapping)?;
+    }
+    transport.reset_target(reset_delay, true)?;
+    // we want to hold the strapping configuration here because in some life cycle states,
+    // the tap multiplexing is dynamic so remove the tap strap would actually change the tap
+    Ok(())
+}
```

Comment:
Yes I copied those from your tests to help debug stuff :) We want to move `reset()` to the lib. Regarding `ScopedBitstream`, I think we ideally want to find what the problem is and not rely on it unless absolutely necessary.

Reply:
I think ScopedBistream is just a [clever trick](https://doc.rust-lang.org/rust-by-example/scope/raii.html) to trigger a bitstream clearing when `_scoped_bitstream` goes out of scope.

Since this is a test, I think we should be explicit about our actions, and just call
```rust
self.transport.dispatch(&ClearBitstream {}); 
```
at the end of the test if our goal is to leave the FPGA in a cleared state. Though I am not sure it is necessary for this test, since `init_target` [already loads a new bitstream](https://cs.opensource.google/opentitan/opentitan/+/master:sw/host/opentitanlib/src/test_utils/init.rs;drc=aa1808981c4da762ace8d7c0ed6b0b208aee60bf;l=138) when tests start. I.e., I think we can just delete all of the ScopedBitsream stuff from this test.